core/src/main/java/org/apache/calcite/rel/rules/SubQueryRemoveRule.java

```diff
@@ -309,8 +311,22 @@ private RexNode rewriteExists(RexSubQuery e, Set<CorrelationId> variablesSet,
     }
 
     builder.as("dt");
+    boolean generateNullsOnRight = true;
+    if (e.rel instanceof LogicalAggregate) {
+      // SELECT f0, count(*) FROM t GROUP BY () will always return 1 row.
+      // That means, the RHS never generates null.
+      final LogicalAggregate aggregate = (LogicalAggregate) e.rel;
+      if (aggregate.getGroupSet().isEmpty()
+            && aggregate.getAggCallList().stream()
+                .anyMatch(c -> c.getAggregation() instanceof SqlCountAggFunction)) {
```

Comment:
Yeah, but only count agg doesn't output nulls.

Reply:
You should be able to trust the aggregate function to derive its own nullability.

For example, SUM knows that even when applied to a not-null column x, if the groupSet is empty it may return null. COUNT knows that it never returns null. 

So maybe just look at the rowType of Aggregate, and look at whether the columns are NOT NULL.

Reply:
Thanks, Julian, let me have a try :)